src/Fixer/ClassNotation/StaticPrivateMethodFixer.php

```diff
@@ -0,0 +1,288 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ClassNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Filippo Tessarotto <zoeslam@gmail.com>
+ */
+final class StaticPrivateMethodFixer extends AbstractFixer
+{
+    /**
+     * @var array
+     */
+    private $magicNames = [
+        '__call' => true,
+        '__callstatic' => true,
+        '__clone' => true,
+        '__construct' => true,
+        '__debuginfo' => true,
+        '__destruct' => true,
+        '__get' => true,
+        '__invoke' => true,
+        '__isset' => true,
+        '__serialize' => true,
+        '__set' => true,
+        '__set_state' => true,
+        '__sleep' => true,
+        '__tostring' => true,
+        '__unserialize' => true,
+        '__unset' => true,
+        '__wakeup' => true,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Converts private methods to `static` where possible.',
+            [
+                new CodeSample(
+                    '<?php
+class Foo
+{
+    public function bar()
+    {
+        return $this->baz();
+    }
+
+    private function baz()
+    {
+        return 1;
+    }
+}
+'
+                ),
+            ],
+            null,
+            'Risky when method contains dynamic generated calls to the instance, or the method is dynamically referenced.'
+        );
+    }
+
+    public function getPriority()
+    {
+        // Must run after ProtectedToPrivateFixer, before StaticLambdaFixer
+        return 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAllTokenKindsFound([T_CLASS, T_PRIVATE]);
```

Comment:
I think the fixer can also be applied inside of Traits

Reply:
Mmm, I think not
```php
// file Foo.php
trait Foo
{
    private function bar() {}
}
// file Baz.php
class Baz
{
    use Foo;
    
    public function wut()
    {
        return $this->bar();
    }
}
```
I cannot fix Foo::bar because I would break Baz::wut behavior; well, not entirely break as php still run without issue, but a static analyzer would complain

Reply:
I'm sorry I just read https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4557#discussion_r326203871: yes we can fix Traits too pointing the possible side effects into the risky description.

But honestly the two scenario are much differently spread imho:

1. A class that use a private method of a trait: ok, bad but common, we can't use this fixer for Trait as we would (logically) break the code
1. A trait that use a private method of a class: WUT? Are you kidding me? Yeah it's feasible, but if you write crappy code it's not our fault

I feel safe to fix classes and not fix traits, what do you think?

Reply:
Seems like a good idea.

I don't think it will be a common problem, but it is one of those nasty edge cases.